src/core/geometry/qgsabstractgeometry.h

```diff
@@ -484,4 +484,12 @@ struct CORE_EXPORT QgsVertexId
   VertexType type;
 };
 
+#ifndef SIP_RUN
+template <class T>
+inline T qgsgeometry_cast( const QgsAbstractGeometry *geom )
+{
+  return const_cast<T>( reinterpret_cast<T>( 0 )->cast( geom ) );
```

Comment:
I think there's not much to be typechecked or safe at any time with casting a `nullptr` I guess :P